sdks/java/core/src/main/java/org/apache/beam/sdk/util/gcsfs/GcsPath.java

```diff
@@ -601,6 +600,27 @@ public String toResourceName() {
     return sb.toString();
   }
 
+  private static final Pattern HAS_GLOB = Pattern.compile(".*(\\[|\\*|\\?).*");
+  public String getBrowseUrl() {
+    StringBuilder builder = new StringBuilder("https://console.cloud.google.com/storage/browser/");
```

Comment:
It appears these URLs work for browsing to "buckets" (i.e. directories), but not "objects" (i.e. files). There's another URL schema, "https://storage.cloud.google.com/*" which works for exactly the opposite-- buckets but not objects.  Without doing a query, we won't always be able to tell what's a bucket and what's an object.